api/routes/user.route.js

```diff
@@ -71,4 +71,21 @@ userRoutes.route('/deleteUser/:username').get(function(req, res) {
     });
 });
 
+// Update username by id 
+userRoutes.route('/updateUsername/:id').post((req, res) => {
+  console.log(req.body.username);
+    User.findById(req.params.id, (err, user) => {
+        if(user) {
+            console.log("new username: ", req.body.username);
+            user.username = req.body.username;
+            user.save().then(() => {
+              console.log("username now: ", user.username);
+                res.status(200).json("Username updated successfully");
+            }).catch(err => {
+                res.status(404).send(err);
+            })
+        }
```

Comment:
The `404` should probably be for if the user is 'not found' (`!user`), with like a `500` if some error occurs while `save()`ing. 

Wikipedia gives [a reasonable-looking list of the HTTP status codes](https://en.wikipedia.org/wiki/List_of_HTTP_status_codes). I'd say I've used them wrong in plenty of places in the API but might as well start using them properly! 🤣

Maybe something like:
```suggestion
             }).catch(err => {
                res.status(500).send(err);
            })
        } else {
            res.status(404).send(`User with _id ${req.params.id} could not be found`);
        }
```

Reply:
I have a running list of "structure inconsistencies" throughout scealai, maybe we can have a meeting to discuss which format/way would be best for handling errors, status codes, etc.?  (Yeah a lot of these functions were copy/paste :) )